### PR TITLE
Fix CI integration live issue

### LINKED ISSUE #491: CI - DB Integration - Live  is not running tests
I just noticed our CI integration is not running the actual tests but just prints the sqlachemy version (live and local)

image

It should be something like this (see the running time and the output)

image

My guess is it's something with the syntax which runs only the first part and ignores the second

```
python -c "import sqlalchemy; print('Sqlalchemy version ', sqlalchemy.__version__)"
pytest src/tests/integration --verbose
```

cc: @idomic @edublancas

### LINKED ISSUE #495: Unit Test Failing since duckdb 0.8.0
We will fail the unit test since duckdb just release their new version, 0.8.0

We need to downgrade the duckdb version.

I found this issue while doing the https://github.com/ploomber/jupysql/pull/493

[duckdb release history](https://pypi.org/project/duckdb/#history)

### PR DESCRIPTION
**Pull request by tonykploomber**
## Describe your changes

- Fix by adding the pipeline symbol `|` on the `run` attribute
Ref: https://stackoverflow.com/questions/56726429/how-to-run-multiple-commands-in-one-github-actions-docker

- Fix the unit test failing issue #491
- Fix the duckdb 0.8.0 issue #495 #498
- Skip the test case `test_profile_query` in `test_generic_db_operations.py` since @yafimvo found it will generate unclosed session to block the tear_down process in the testing part, causing the timeout issue

## Issue number

Closes #495
Closes #491 

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--493.org.readthedocs.build/en/493/

### CODE REVIEW

**Comment by tonykploomber** [comment]
@yafimvo 
Now I believe it's running the integration testing for live databases, please review!

Screenshot 2023-05-17 at 1 10 02 PM

So now we have two integration testing 
- non-live, which starts different database docker containers, it's using SQLAlchemy 2.0+  
- live, which tests on snowflake online database, it's using SQLAlchemy 1.4

**Comment by yafimvo** [comment]
@tonykploomber I see the [non-live test](https://github.com/ploomber/jupysql/actions/runs/4999503073/jobs/8955955613?pr=493#step:5:1) is still not running the actual tests

image


> * live, which tests on snowflake online database, it's using SQLAlchemy 1.4

I think downgrading to 1.4 might be a bit problematic since most of the DBs support SQLAlchemy > 2 and we need to test them (even though we're testing them locally).

Is it possible to define 2 different jobs, one for SQLAlchemy > 2 (and exclude Snowflake) and one for 1.4?

**Comment by tonykploomber** [comment]
> @tonykploomber I see the [non-live test](https://github.com/ploomber/jupysql/actions/runs/4999503073/jobs/8955955613?pr=493#step:5:1) is still not running the actual tests
> 
> image
> 
> > * live, which tests on snowflake online database, it's using SQLAlchemy 1.4
> 
> I think downgrading to 1.4 might be a bit problematic since most of the DBs support SQLAlchemy > 2 and we need to test them (even though we're testing them locally).
> 
> Is it possible to define 2 different jobs, one for SQLAlchemy > 2 (and exclude Snowflake) and one for 1.4?

Actually we are doing the thing as you mentioned. We have two different jobs now:
- live: only runs snowflake integration testing, which uses 1.4
- non-live: runs all testings except the snowflake part, which uses 2+

Now we separate those integration testing by it's database destination.
In the future we might have a case that uses an live database service and also needs SQLalchemy 2+ to run it, but we can do that once we met that

**Comment by tonykploomber** [comment]
Let's review again, I also address this https://github.com/ploomber/jupysql/issues/495

**Comment by yafimvo** [comment]
@tonykploomber It seems something is wrong with the [CI - DB Integration - Local](https://github.com/ploomber/jupysql/actions/runs/5010200530/jobs/8979856162?pr=493) test. It has been running for 3 hours and is still ongoing.

**Comment by tonykploomber** [comment]
Update:
I think the root cause should be [this](https://docs.sqlalchemy.org/en/20/faq/metadata_schema.html#my-program-is-hanging-when-i-say-table-drop-metadata-drop-all)

We have a `tear_down_generic_testing_data` function that drops all testing table data, in tear_down section.

Something wrong with `mysql` and `mariadb` that has a connection (or session) blocking the drop command from another session. 

Screenshot 2023-05-19 at 10 30 33 PM

As snapshot, we can see it's hanging on those databases

@yafimvo, I tried to add `close_all_sessions` before calling the Base.metadata.drop_all (this causes hanging).
But it doesn't work, still need more investigation
```
def drop_table(engine, table_name):
    Base = declarative_base()
    metadata = MetaData()
    metadata.reflect(bind=engine)
    table = metadata.tables[table_name]
    if table is not None:
        close_all_sessions()
        Base.metadata.drop_all(engine, [table], checkfirst=True)
        print ("Clear done: ", table)
```

**Comment by tonykploomber** [comment]
This is the related discussion

**Comment by tonykploomber** [comment]
@yafimvo Let's review again, now I skip the `test_profile_query` case by adding `pytest.skip("Skip on unclosed session issue")
`

File: setup.py [hunk 1]
```diff
@@ -54,6 +54,8 @@
     "dockerctx",
     "pyarrow",
     "psycopg2-binary",
+    "psycopg-binary",
+    "psycopg-pool",
```
Comment (yafimvo) [inline setup.py:58]: We added it for debugging when we thought it was questdb. I think we can keep it as it was before.

File: src/sql/_testing.py [hunk 2]
```diff
@@ -316,7 +316,7 @@ def mssql(is_bypass_init=False):
                 "MSSQL_SA_PASSWORD": db_config["password"],
                 "ACCEPT_EULA": "Y",
             },
-            ready_test=lambda: database_ready(database="MSSQL"),
+            ready_test=lambda: database_ready(database="MSSQL", timeout=60),
```
Comment (yafimvo) [inline src/sql/_testing.py:319]: I think we can add this timeout for every database not just "MSSQL"

File: src/tests/integration/conftest.py [hunk 3]
```diff
@@ -55,12 +54,8 @@ def test_table_name_dict():
 
 
 def drop_table(engine, table_name):
-    Base = declarative_base()
-    metadata = MetaData()
-    metadata.reflect(bind=engine)
-    table = metadata.tables[table_name]
-    if table is not None:
-        Base.metadata.drop_all(engine, [table], checkfirst=True)
+    tbl = Table(table_name, MetaData(), autoload_with=engine)
```
Comment (yafimvo) [inline src/tests/integration/conftest.py:57]: can you briefly explain what did you do here?

**Review by yafimvo — CHANGES_REQUESTED** [review]
@tonykploomber 
Added some minor notes. Let's address them and merge it.

Did you open an issue about `test_profile_query`?

File: setup.py (hunk 1 above)
Comment (tonykploomber) [reply setup.py:58]: Got it!

File: src/sql/_testing.py (hunk 2 above)
Comment (tonykploomber) [reply src/sql/_testing.py]: fix in [1ae8673](https://github.com/ploomber/jupysql/pull/493/commits/1ae867365472eedb110e4bbf9ee42f7f5ba53a97)

File: setup.py (hunk 1 above)
Comment (tonykploomber) [reply setup.py]: fix in [1ae8673](https://github.com/ploomber/jupysql/pull/493/commits/1ae867365472eedb110e4bbf9ee42f7f5ba53a97)

**Review by yafimvo — APPROVED** [review]

**Comment by yafimvo** [comment]
@tonykploomber Good job!

**Pull request merged**